src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -905,6 +923,36 @@ await CreateCodeFlowPullRequestAsync(
         {
             await UpdateAssetsAndSources(update, pr);
         }
+        catch (ConflictInPrBranchException conflictException)
+        {
+            // The PR we're trying to update has a conflict with the source repo. We will mark it as blocked, not allowing any updates from this
+            // subscription till it's merged, or the conflict resolved. We'll set a reminder to check on it.
+            StringBuilder sb = new();
+            sb.AppendLine($"There was a conflict in the PR branch when flowing source from {update.GetRepoAtCommitUri()}");
+            sb.AppendLine("Files conflicting with the head branch:");
+            foreach (var file in conflictException.FilesInConflict)
+            {
+                sb.AppendLine($" - [{file}]({update.GetFileAtCommitUri(file)})");
+            }
+            sb.AppendLine();
+            sb.AppendLine("Updates from this subscription will be blocked until the conflict is resolved, or the PR is merged");
+
+            (var targetRepository, _) = await GetTargetAsync();
+            var remote = await _remoteFactory.CreateRemoteAsync(targetRepository);
+
+            await remote.CommentPullRequestAsync(pr.Url, sb.ToString());
```

Comment:
Posting the comment should be try/catched in case it fails for whatever reason so we update the cache still. Or it can be done at the end.

Reply:
good point, the remote.GetLatestCommitAsync() too, if that fails it's probably better to keep the same sha, than to not put a reminder at all